javascript/ql/src/semmle/javascript/frameworks/Vuex.qll

```diff
@@ -0,0 +1,436 @@
+    }
+
+    /** Gets the Vue component in which the generated functions are installed. */
+    VueAPI::VueConfigObject getVueConfigObject() {
+      exists(DataFlow::ObjectLiteralNode obj |
+        obj.getASpreadProperty() = getReturn().getAUse() and
+        result.getAMember().getARhs() = obj
+      )
+      or
+      result.getAMember().getARhs() = this
+    }
+  }
+
+  /**
+   * Gets an API node that refers to a property of a Vue component instance,
+   * which has been bound by a helper of kind `helperName` to something named `storeName`.
+   *
+   * For example, `mapGetters({foo: 'bar})` will cause `this.foo` in the affected Vue component
+   * to be returned by `getAMappedAccess("mapGetters", "bar")`.
+   */
+  API::Node getAMappedAccess(string helperName, string storeName) {
+    exists(MapHelperCall call, string localName |
+      call.getHelperName() = helperName and
+      call.hasMapping(localName, storeName) and
+      result = call.getVueConfigObject().getAnInstanceRef().getMember(localName) and
+      localName != "*"
+    )
+  }
+
+  // -----------------------------------------------------------------------------
+  // Flow from getter method return value to getter call
+  // -----------------------------------------------------------------------------
+  /** Gets a value that is returned by a getter registered with the given name. */
+  private DataFlow::Node getterPred(string name) {
+    exists(string prefix, string prop |
+      result = storeConfigObject(prefix).getMember("getters").getMember(prop).getReturn().getARhs() and
+      name = prefix + prop
+    )
+  }
+
+  /** Gets a property access that may receive the produced by a getter of the given name. */
+  private DataFlow::Node getterSucc(string name) {
+    exists(string prefix, string prop |
+      result = storeRef(prefix).getMember("getters").getMember(prop).getAnImmediateUse() and
+      prop != "*" and
+      name = prefix + prop
+    )
+    or
+    result = getAMappedAccess("mapGetters", name).getAnImmediateUse()
+  }
+
+  /** Holds if `pred -> succ` is a step from a getter function to a relevant property access. */
+  private predicate getterStep(DataFlow::Node pred, DataFlow::Node succ) {
+    exists(string name |
+      pred = getterPred(name) and
+      succ = getterSucc(name)
+    )
+  }
+
+  // -----------------------------------------------------------------------------
+  // --- Flow from commit() to mutation method, or dispatch() to action method ---
+  // -----------------------------------------------------------------------------
+  /**
+   * Gets a node that refers to the `commit` or `dispatch` function of the store.
+   *
+   * For our purposes, the (`commit`, `mutations`, `mapMutations`) properties
+   * behave in nearly the same way as (`dispatch`, `actions`, `mapActions`) so we
+   * model them together, with `kind` being either `commit` or `dispatch`.
+   */
+  API::Node commitLikeFunctionRef(string kind, string prefix) {
```

Comment:
```suggestion
  API::Node commitLikeCallee(string kind, string prefix) {
```

(Very optional) 

How about `callee` instead of `functionRef`?

Reply:
I prefer `functionRef`. To me, `callee` sounds like we're tracking the callee of a specific call site.